crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -0,0 +1,170 @@
+use super::Primitive2d;
+use crate::Vec2;
+
+/// A normalized vector pointing in a direction in 2D space
+#[derive(Clone, Copy, Debug)]
+pub struct Direction2d(Vec2);
+
+impl From<Vec2> for Direction2d {
+    fn from(value: Vec2) -> Self {
+        Self(value.normalize())
+    }
+}
+
+impl Direction2d {
+    /// Create a direction from a [Vec2] that is already normalized
+    pub fn from_normalized(value: Vec2) -> Self {
+        debug_assert!(value.is_normalized());
+        Self(value)
+    }
+}
+
+impl std::ops::Deref for Direction2d {
+    type Target = Vec2;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+/// A circle primitive
+#[derive(Clone, Copy, Debug)]
+pub struct Circle {
+    /// The radius of the circle
+    pub radius: f32,
+}
+impl Primitive2d for Circle {}
+
+/// An unbounded plane in 2D space. It forms a separating surface trough the origin,
+/// stretching infinitely far
+#[derive(Clone, Copy, Debug)]
+pub struct Plane2d {
+    /// The normal of the plane, the plane will be placed perpendicular to this direction
+    pub normal: Direction2d,
+}
+impl Primitive2d for Plane2d {}
+
+/// An infinite line along a direction in 2D space.
+///
+/// For a finite line: [`LineSegment2d`]
+#[derive(Clone, Copy, Debug)]
+pub struct Line2d {
+    /// The direction of the line, the line extends infinitely in both the positive
+    /// and negative of this direction
+    pub direction: Direction2d,
+}
+impl Primitive2d for Line2d {}
+
+/// A section of a line along a direction in 2D space.
+#[derive(Clone, Debug)]
+pub struct LineSegment2d {
+    /// The direction of the line
+    pub direction: Direction2d,
+    /// Half the length of the line segment, the segment extends by this amount in both
+    /// the positive and negative direction
+    pub half_length: f32,
+}
+impl Primitive2d for LineSegment2d {}
+
+impl LineSegment2d {
+    /// Get a line segment and translation from a start and end position of a line segment
+    ///
+    /// Panics if start == end
+    pub fn from_start_end(start: Vec2, end: Vec2) -> (Self, Vec2) {
```

Comment:
I don't think `new` would be the right name here, since you could also construct a line from a direction and full length. I guess I could rename it to get rid of the `start` and `end` tho, alice commented about these names being confusing in this contacts before too :thinking:. If we create PartialEq implementations (probably for a later PR?) there's a few cases like this that should be handled I think